pychunkedgraph/graph/subgraph.py

```diff
@@ -0,0 +1,232 @@
+from typing import List
+from typing import Dict
+from typing import Tuple
+from typing import Union
+from typing import Iterable
+from typing import Sequence
+from typing import Optional
+
+from numpy import uint64
+from numpy import array
+from numpy import concatenate
+
+from .edges import Edges
+from .chunks.utils import normalize_bounding_box
+
+
+class SubgraphProgress:
+    """
+    Helper class to keep track of node relationships
+    while calling cg.get_subgraph(node_ids)
+    """
+
+    def __init__(self, meta, node_ids, return_layers, serializable):
+        self.meta = meta
+        self.node_ids = node_ids
+        self.return_layers = return_layers
+        self.serializable = serializable
+
+        self.node_to_subgraph = {}
```

Comment:
suggestion to make this a `collections.defaultdict(list)` to save the hassle to create lists in the code below

Reply:
I think this was written by @manuel-castro 
@manuel-castro what do you think? I will make the changes but not sure if it breaks anything, can you take a look and let me know?

Reply:
self.node_to_subgraph is a dict that contains dicts that contains lists, so collections.defaultdict(list) wouldn't work, is collections.defaultdict(collections.defaultdict(list)) a thing?

Reply:
not sure but if it isn't straightforward we should probably leave it as is for now

Reply:
`self.node_to_subgraph = defaultdict(lambda: defaultdict(list))`

Reply:
seems to be working, tests passed